storage/redis_storage.go

```diff
@@ -127,7 +128,34 @@ func (storage *RedisStorage) WriteReportForCluster(
 		return err
 	}
 
+	data := ctypes.SimplifiedReport{
```

Comment:
Maybe we should do everything in this method within a transaction?

https://redis.uptrace.dev/guide/go-redis-pipelines.html#transactions
https://pkg.go.dev/github.com/go-redis/redis?utm_source=godoc#Client.Watch